src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -179,25 +197,40 @@ def cli_cosmosdb_update(client,
     if enable_public_network is not None:
         public_network_access = 'Enabled' if enable_public_network else 'Disabled'
 
+    _ip_rules = None
+    if ip_rules is not None:
+        _ip_rules = get_ip_rules(ip_rules)
+
     params = DatabaseAccountUpdateParameters(
         locations=locations,
         tags=tags,
         consistency_policy=consistency_policy,
-        ip_range_filter=ip_range_filter,
+        ip_rules=_ip_rules,
         is_virtual_network_filter_enabled=enable_virtual_network,
         enable_automatic_failover=enable_automatic_failover,
         capabilities=capabilities,
         virtual_network_rules=virtual_network_rules,
         enable_multiple_write_locations=enable_multiple_write_locations,
         disable_key_based_metadata_write_access=disable_key_based_metadata_write_access,
-        public_network_access=public_network_access)
+        public_network_access=public_network_access,
+        enable_analytical_storage=enable_analytical_storage)
 
     async_docdb_update = client.update(resource_group_name, account_name, params)
     docdb_account = async_docdb_update.result()
     docdb_account = client.get(resource_group_name, account_name)  # Workaround
     return docdb_account
 
 
+def get_ip_rules(ip_rules):
+    _ip_rules = []
+    if ip_rules:
+        for ip_address in ip_rules:
+            _ip_address_or_range = {}
```

Comment:
See how we do this for capability objects here: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/cosmosdb/_validators.py#L38

The python library should have IpAddressOrRange object you can use instead of hard-coding a dictionary here.

Reply:
Done